array.c

```diff
@@ -7920,11 +7923,27 @@ rb_ary_deconstruct(VALUE ary)
  *  - Indexes -1 through -3 are in range.
  *  - Index -4 is out of range.
  *
+ *  == \Integer-Convertible Arguments
+ *
  *  Although the effective index into an array is always an integer,
  *  some methods (both within and outside of class \Array)
- *  accept one or more non-integer arguments that are
+ *  accept one or more arguments that are
  *  {integer-convertible objects}[rdoc-ref:implicit_conversion.rdoc@Integer-Convertible+Objects].
+ *  Thus the given argument may be any Numeric
+ *  that can be converted to an integer:
+ *
+ *  - Integer (of course).
```

Comment:
I suggest leaving out the "of course" part. It could sound sarcastic to the reader.

Reply:
Omitted (and one other small change).